src/main/java/com/aol/cyclops/types/Value.java

```diff
@@ -49,6 +51,29 @@
 @FunctionalInterface
 public interface Value<T> extends Supplier<T>, Foldable<T>, Convertable<T>, Publisher<T>, Predicate<T>, Zippable<T> {
 
+	/**
+     * Perform a flatMap operation that will work for Value and take the first value from the Iterable returned.
+     * 
+     * @param fn FlatMaping function
+     * @return Value with flattening transformation
+     */
+    default <R> Value<R> flatMapFirst(Function<? super T, ? extends Iterable<? extends R>> fn) {
```

Comment:
Hi @johnmcclean-aol , ok. Let me check out the MonadicValue class/interface and move the implementation over to it. Thanks for your comment.